### ISSUE TITLE
Home Assistant Showing Wrong State

### ISSUE THREAD

**Issue by Taseen18** [guerrerotook/securitas-direct-new-api#2]
When i "arm home" through home assistant the state of the alarm updates as "armed away" rather than "armed home"

**Comment by Taseen18** [comment]
Immediately after i arm it sometimes it changes back to "disarmed" on home assistant but in reality it will remain as armed

**Comment by guerrerotook** [comment]
For the second comment, there is a delay if you change the alarm with you key. I only refresh state every 5 minutes.

**Comment by siom7** [comment]
Same as @Taseen18 for me, when I armed night my alarm is turning on Armed Away and when I armed away nothing happend.

**Comment by Javvaz** [comment]
it doesn't identify properly the status alarm. It should identify properly all combinations among away, night, home, perimetral. This issue provoke that other automations cannot properly know the status alarm and they can be executed in case of status change.
My propose is create enough status in order to encompass all combinations: away, away+perimetral, night, night+perimetral, home, home+perimetral, only perimetral, and finally disarm

**Comment by emijime** [comment]
The same happens to me. The status is not correctly shown in the homeassistant panel when you perform any change for example through verisure application.

**Comment by bvis** [comment]
What I’ve seen is that the mistaken status is produced when you press the desired state in the HA alarm panel.

**Comment by Xeovar** [comment]
I've decided to reuse this issue rather than create a new one.
Just installed this integration (Portugal) and it sort of works. Sort of, because in Portugal the response time of the API is extremely slow, even on the official app. Can you please add a delay between alarm state change request and get current alarm state requests? Like at least 30 seconds.... Or maybe run two get current state requests one after another? First just after arm command the other 30 sec later to force quicker update than 5 min?

**Comment by guerrerotook** [comment]
> I've decided to reuse this issue rather than create a new one. Just installed this integration (Portugal) and it sort of works. Sort of, because in Portugal the response time of the API is extremely slow, even on the official app. Can you please add a delay between alarm state change request and get current alarm state requests? Like at least 30 seconds.... Or maybe run two get current state requests one after another? First just after arm command the other 30 sec later to force quicker update than 5 min?

What you mention about the _Can you please add a delay between alarm state change request and get current alarm state requests? _ it has been changed so when you make an arm or disarm request, there isn't a update state request after. Just the operation and that is.

**Comment by Xeovar** [comment]
> > I've decided to reuse this issue rather than create a new one. Just installed this integration (Portugal) and it sort of works. Sort of, because in Portugal the response time of the API is extremely slow, even on the official app. Can you please add a delay between alarm state change request and get current alarm state requests? Like at least 30 seconds.... Or maybe run two get current state requests one after another? First just after arm command the other 30 sec later to force quicker update than 5 min?
> 
> What you mention about the _Can you please add a delay between alarm state change request and get current alarm state requests? _ it has been changed so when you make an arm or disarm request, there isn't a update state request after. Just the operation and that is.

Thats not what my alarm log indicates:
![image](https://user-images.githubusercontent.com/65547024/156810183-3c321e31-6144-4882-a3d0-d84637d80b25.png)

Immediately after I press arm home on the alarm card (using the standard one), the request is sent, but the state of alarm component in HA doesnt change (on verisure app side it already armed). Only when I trigger refresh by arming again the component in HA refreshes and shows proper state.

**Comment by guerrerotook** [comment]
Check the new setting in the configuration.

https://github.com/guerrerotook/securitas-direct-new-api/blob/main/README.md

There is a new setting check_alarm_panel that check the last version of the alarm status instead of getting that from the alarm.

Also, did you update to the latest version?

**Comment by Xeovar** [comment]
I just installed via HACS, it is most recent version. I havent used the check_alarm_panel switch, so I assume it would be default true. Ill test with it set to false and see what happens.
Im sure that this request goes out and it returns too early, before the alarm arms, because in the developer tools/states I see this:
message: O seu alarme está desligado
with the timestamp almost the same as the moment I sent the arm alarm request.
Thanks for your help :)

**Comment by guerrerotook** [comment]
> I just installed via HACS, it is most recent version. I havent used the check_alarm_panel switch, so I assume it would be default true. Ill test with it set to false and see what happens. Im sure that this request goes out and it returns too early, before the alarm arms, because in the developer tools/states I see this: message: O seu alarme está desligado with the timestamp almost the same as the moment I sent the arm alarm request. Thanks for your help :)

Yes, you need to set to false.

**Comment by Xeovar** [comment]
Ok I have set the check_alarm_panel switch to false and it now works perfectly.
Can you just tell me if I can turn off the code requirement on the panel?

**Comment by guerrerotook** [comment]
> Ok I have set the check_alarm_panel switch to false and it now works perfectly. Can you just tell me if I can turn off the code requirement on the panel?

Yes, you simply remove the code from the configuration I guess. This code is local to Home Assistant.

**Comment by Xeovar** [comment]
Thank you for help and great work on this thing, been waiting for a working integration over a year...

**Comment by siom7** [comment]
Hi, I tried your component many times but unfortunately my Verisure Hub still crash and I had to call Verisure to reset it… I tried false or true for option check_alarm_panel but still the same… I dont understand why I’m the only with this issue 😔

**Comment by Javvaz** [comment]
I am afraid that the interfaces is again failed and doesn't update the status of alarm until the Securitas APP update its status alarm.
